### ISSUE TITLE
Production action is optional for each factory

### ISSUE THREAD

**Issue by dmcarmo** [Thrillberg/imperial#305]
Each factory **may** produce an army or a fleet, as such we should display all the valid factories (not occupied), select which to produce and add a button to end the production.

**Comment by Thrillberg** [comment]
I agree with this reading of the rules but I also feel like this is a low priority for now since it seems rare that someone would want to waive production at a factory. We should de-prioritize this issue, imo, unless someone feels differently.

**Comment by dmcarmo** [comment]
I thought the same but, as an example, Russia may want to skip production of naval units in the black sea as it's the only nation that can produce there or, more generically, you may not want to produce all the units so that if you use taxation later you can win more taxes (by having less armies than with a full production).

**Comment by Thrillberg** [comment]
I don't understand the Russia Black Sea example but I do agree that for taxation reasons, it seems possible that someone wouldn't want to build a unit.

**Comment by dmcarmo** [comment]
Sorry, ignore the example about Russian / Black Sea, I thought they couldn't exit that area into the Mediterranean :facepalm:
